### ISSUE TITLE
Error in fit method

### ISSUE THREAD

**Issue by Florro** [apache/incubator-mxnet#1307]
The training loss seems to depend on the use of an evaldata set. 

model.fit(X=data_train, eval_data = val_train, eval_metric = mx.metric.np(CRPS) )

produces sometimes significant different training - loss than

model.fit(X=data_train, eval_metric = mx.metric.np(CRPS) )

The Bug is hard to reproduce, and seems to occure with CSVIter. Could be an Error in the ThreadIter.

**Comment by jermainewang** [comment]
Hmm, do you observe this on all kinds of dataset? or it just happens on certain dataset?

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
